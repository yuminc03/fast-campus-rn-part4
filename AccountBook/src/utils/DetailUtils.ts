export const convertToDateString = (time: number) => {
  const date = new Date(time);
  const addZeroIfOneCharacter = (number: number): string => {
    if (number < 10) {
      return `0${number}`;
    }

    return number.toString();
  };

  return `${date.getFullYear()}-${addZeroIfOneCharacter(
    date.getMonth() + 1,
  )}-${addZeroIfOneCharacter(date.getDate())} ${addZeroIfOneCharacter(
    date.getHours(),
  )}:${addZeroIfOneCharacter(date.getMinutes())}`;
};
