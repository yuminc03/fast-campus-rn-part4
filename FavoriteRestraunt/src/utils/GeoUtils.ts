const KAKAO_API_KEY = 'e9fc7ea9d6089aae5c7aabca5f4f170c';

export const getAddressFromCoords = (
  latitude: number,
  longitude: number,
): Promise<string | null> => {
  return fetch(
    `https://dapi.kakao.com/v2/local/search/address.json?x=${longitude}&y=${latitude}`,
    {
      method: 'GET',
      headers: {
        Authorization: `KakaoAK ${KAKAO_API_KEY}`,
      },
    },
  )
    .then(result => result.json())
    .then(result => {
      console.log(result);

      if (result.meta.total_count === 0) {
        return null;
      }

      if (result.documents.length === 0) {
        return null;
      }

      const addressItem = result.documents[0];
      return addressItem.address.address_name;
    });
};

export const getCoordsFromAddress = (
  address: string,
): Promise<{latitude: number; longitude: number; address: string} | null> => {
  return fetch(
    `https://dapi.kakao.com/v2/local/search/address.json?query=${address}`,
    {
      method: 'GET',
      headers: {
        Authorization: `KakaoAK ${KAKAO_API_KEY}`,
      },
    },
  )
    .then(result => result.json())
    .then(result => {
      console.log(result);

      if (result.meta.total_count === 0) {
        return null;
      }

      if (result.documents.length === 0) {
        return null;
      }

      const addressItem = result.documents[0];
      return {
        latitude: addressItem.y,
        longitude: addressItem.x,
        address: addressItem.address_name,
      };
    });
};

export const getCoordsFromKeyword = (
  keyword: string,
): Promise<{latitude: number; longitude: number; address: string} | null> => {
  return fetch(
    `https://dapi.kakao.com/v2/local/search/keyword.json?query=${keyword}`,
    {
      method: 'GET',
      headers: {
        Authorization: `KakaoAK ${KAKAO_API_KEY}`,
      },
    },
  )
    .then(result => result.json())
    .then(result => {
      console.log(result);

      if (result.meta.total_count === 0) {
        return null;
      }

      if (result.documents.length === 0) {
        return null;
      }

      const addressItem = result.documents[0];
      return {
        latitude: addressItem.y,
        longitude: addressItem.x,
        address: addressItem.address_name,
      };
    });
};
