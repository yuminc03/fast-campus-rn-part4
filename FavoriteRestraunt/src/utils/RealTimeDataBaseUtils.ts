import database from '@react-native-firebase/database';

export const saveNewRestraunt = async (params: {
  title: string;
  address: string;
  latitude: number;
  longitude: number;
}) => {
  const db = database().ref('/restraunt');
  const saveItem = {
    title: params.title,
    address: params.address,
    latitude: params.latitude,
    longitude: params.longitude,
  };

  await db.push().set({
    ...saveItem,
  });
};
