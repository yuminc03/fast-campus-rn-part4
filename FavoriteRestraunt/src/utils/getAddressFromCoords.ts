const KAKAO_API_KEY = 'e9fc7ea9d6089aae5c7aabca5f4f170c';

export const getAddressFromCoords = (
  latitude: number,
  longitude: number,
): Promise<string | null> => {
  return fetch(
    `https://dapi.kakao.com/v2/local/search/address.json?x=${longitude}&y=${latitude}`,
    {
      method: 'GET',
      headers: {
        Authorization: `KakaoAK ${KAKAO_API_KEY}`,
      },
    },
  )
    .then(result => result.json())
    .then(result => {
      console.log(result);

      if (result.meta.total_count === 0) {
        return null;
      }

      if (result.documents.length === 0) {
        return null;
      }

      const addressItem = result.documents[0];
      return addressItem.address.address_name;
    });
};
