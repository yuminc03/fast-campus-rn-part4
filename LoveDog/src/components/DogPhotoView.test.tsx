import {render, screen, waitFor} from '@testing-library/react-native';
import MockAdapter from 'axios-mock-adapter';

import {DogPhotoView} from './DogPhotoView';
import {api} from '../utils/AxiosUtils';

describe('DogPhoto Render Test', () => {
  let mock: MockAdapter;

  // 모든 테스트가 돌기 전 실행됨
  beforeAll(() => {
    mock = new MockAdapter(api);
  });

  // 모든 테스트가 끝났을 때
  afterAll(() => {
    mock.reset();
  });

  test('DogPhotoView Render Snapshot', async () => {
    mock.onGet(new RegExp('/breeds/image/random')).reply(200, {
      message: 'TEST_MESSAGE',
      status: 'SUCCESS',
    });

    const component = render(<DogPhotoView />);
    // 특정 component를 찾을 때까지 기다림 (2초 동안)
    await waitFor(() => component.findByTestId('image'), {timeout: 2000});

    expect(screen.toJSON()).toMatchSnapshot();
  });
});
