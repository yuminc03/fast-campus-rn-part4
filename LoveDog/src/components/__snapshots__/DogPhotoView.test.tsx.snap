// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DogPhoto Render Test DogPhotoView Render Snapshot 1`] = `
<Image
  source={
    {
      "uri": "TEST_MESSAGE",
    }
  }
  style={
    [
      undefined,
      {
        "height": 525,
        "width": 525,
      },
    ]
  }
  testID="image"
/>
`;
